staging/src/k8s.io/apiserver/pkg/apis/apiserverinternal/validation/validation_test.go

```diff
@@ -0,0 +1,259 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"strings"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/apis/apiserverinternal"
+)
+
+func TestValidateServerStorageVersion(t *testing.T) {
+	cases := []struct {
+		ssv         apiserverinternal.ServerStorageVersion
+		expectedErr string
+	}{
+		{
+			ssv: apiserverinternal.ServerStorageVersion{
+				APIServerID:       "-fea",
```

Comment:
Is there a particular reason for this to be illegal?

Reply:
We followed the requirements of pods names.

https://github.com/kubernetes/kubernetes/pull/88607/files#diff-a1a21d2d3712f2e0be9d659a816cdbabR51